core/src/main/scala/kafka/log/Log.scala

```diff
@@ -194,7 +194,7 @@ class Log(val dir: File,
               segment.index.sanityCheck()
           } catch {
             case e: java.lang.IllegalArgumentException =>
-              warn("Found a corrupted index file, %s, deleting and rebuilding index...".format(indexFile.getAbsolutePath))
+              warn("Found a corrupted index file, %s, deleting and rebuilding index.".format(indexFile.getAbsolutePath), e)
```

Comment:
Yes, I think including the message, but not the stacktrace in the exception is enough.